src/content/BatchResponseContent.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module BatchResponseContent
+ */
+
+import { BatchResponseCollection, BatchResponse } from "./BatchItem";
+
+/**
+ * @class
+ * Class that handles BatchResponseContent
+ */
+export class BatchResponseContent {
+  /**
+   * To hold the responses
+   */
+  private readonly responses: Map<string, BatchResponse>;
+
+  /**
+   * @public
+   * @constructor
+   * Creates the BatchResponseContent instance
+   * @param {BatchResponseCollection} response - The response body returned for batch request from server
+   * @returns An instance of a BatchResponseContent
+   */
+  public constructor(response: BatchResponseCollection) {
+    this.responses = new Map();
+    this.update(response);
+  }
+
+  /**
+   * @public
+   * Updates the Batch response content instance with given responses.
+   * @param {BatchResponseCollection} response - The response json representing batch response message
+   * @returns Nothing
+   */
+  public update(response: BatchResponseCollection): void {
+    const responses = response.responses;
+    for (let i = 0, l = responses.length; i < l; i++) {
+      this.responses.set(responses[i].id, responses[i]);
+    }
+  }
+
+  /**
+   * @public
+   * To get the response of a request for a given request id
+   * @param {string} requestId - The request id value
+   * @returns The Response object instance for the particular request
+   */
+  public getResponseById(requestId: string): BatchResponse | undefined {
```

Comment:
I believe we are missing overloads that take in a parsablefactory or type parameter to deserialize the response body to the desired object type. 

https://github.com/microsoftgraph/msgraph-sdk-design/blob/main/content/BatchResponseContent.md#requirements